src/tools/sign-oci-layer-root-hashes/src/sign.rs

```diff
@@ -1,20 +1,30 @@
 use std::{
-    io::Write,
-    path::PathBuf,
-    process::{Command, Stdio},
+    collections::BTreeMap, io::Write,path::PathBuf, process::{Command, Stdio}
 };
 
 use anyhow::{Context, Error};
 use base64::{engine::general_purpose, Engine};
 use futures::future;
-use oci_distribution::manifest::OciImageManifest;
+use oci_client::manifest::{OciDescriptor, OciImageManifest};
+use oci_client::Reference;
 use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
 
 use crate::{registry, utils};
 
 const ROOT_HASH_LABEL: &str = "io.katacontainers.dm-verity.root-hash";
 const ROOT_HASH_SIG_LABEL: &str = "io.katacontainers.dm-verity.root-hash-sig";
 
+const IMAGE_NAME_LABEL: &str = "image.ref.name";
+const IMAGE_LAYER_DIGEST_LABEL: &str = "image.layer.digest";
+const IMAGE_LAYER_ROOT_HASH_LABEL: &str = "image.layer.root_hash";
+const IMAGE_LAYER_SIGNATURE_LABEL: &str = "image.layer.signature";
+const SIGNATURE_ARTIFACT_TYPE: &str = "application/vnd.oci.mt.pkcs7";
+const SIGNATURE_MEDIA_TYPE: &str = "application/vnd.oci.image.layer.v1.erofs.sig";
```

Comment:
Fine for the next PR, would be nice to create a common package between solar and TS that would hold the common constants.

Reply:
I created a common package containing the common constants and use them in solar. They can be also used in tardev-snapshotter, but it'd better to update the tardev-snapshotter PR once this change gets merged to avoid conflicts.